host/xtest/regression_4000.c

```diff
@@ -4747,6 +4813,38 @@ static void xtest_tee_test_4007_x25519(ADBG_Case_t *c)
 ADBG_CASE_DEFINE(regression, 4007_x25519, xtest_tee_test_4007_x25519,
 		"Test TEE Internal API Generate X25519 key");
 
+
+static void xtest_tee_test_4007_ed25519(ADBG_Case_t *c)
+{
+	TEEC_Session session = { };
+	uint32_t ret_orig = 0;
+
+	if (!ADBG_EXPECT_TEEC_SUCCESS(c,
+			xtest_teec_open_session(&session, &crypt_user_ta_uuid,
+						NULL, &ret_orig)))
+		return;
+
+	if (!ta_crypt_cmd_is_algo_supported(c, &session, TEE_ALG_ED25519,
+					    TEE_ECC_CURVE_25519)) {
+		Do_ADBG_Log("ED25519 not supported: skip subcase");
+		goto out;
+	}
+
+	Do_ADBG_BeginSubCase(c, "Generate Ed25519 key");
+
+	if (!ADBG_EXPECT_TRUE(c,
+			generate_and_test_key(c, &session,
+					      TEE_TYPE_ED25519_KEYPAIR, 0, 256,
+					      NULL, 0)))
+		return;
```

Comment:
shouold remove instruction `return` here:

```c
	Do_ADBG_BeginSubCase(c, "Generate Ed25519 key");

	ADBG_EXPECT_TRUE(c, generate_and_test_key(c, &session,
						   TEE_TYPE_ED25519_KEYPAIR,
						   0, 256, NULL, 0));

	Do_ADBG_EndSubCase(c, "Generate Ed25519 key");
out:
	TEEC_CloseSession(&session);

Reply:
done, thank you!